presto-hive/src/main/java/io/prestosql/plugin/hive/HiveSplitManager.java

```diff
@@ -172,6 +174,16 @@ public ConnectorSplitSource getSplits(
             SplitSchedulingStrategy splitSchedulingStrategy)
     {
         HiveTableHandle hiveTable = (HiveTableHandle) tableHandle;
+        if (HiveSessionProperties.isQueryPartitionFilterRequired(session)) {
+            List<HiveColumnHandle> partitionColumns = hiveTable.getPartitionColumns();
+            if (!partitionColumns.isEmpty() && hiveTable.getEnforcedConstraint().isAll()) {
+                String partitionColumnNames = partitionColumns.stream().map(n -> n.getName()).collect(Collectors.joining(","));
+                throw new PrestoException(
+                    StandardErrorCode.QUERY_REJECTED,
+                    String.format("Filter required on %s.%s for at least one partition column: %s ", hiveTable.getSchemaName(), hiveTable.getTableName(), partitionColumnNames));
+            }
+        }
+
```

Comment:
Doing this check in getSplits() doesn't work well because EXPLAIN won't execute it (and thus will pass while query will fail). How about a new Connector method like this?

    /**
     * Allow the connector to validate the logical plan of the query to enable connector specific
    *  validation to occur.
    */
    default void validate(Plan logicalPlan) {}

Reply:
That could reasonable API to add. We've talked about adding explicit lifecycle steps before and after a query executes, but never got around to finalizing a design. Such a thing would help remove some "hacks" like `BeginTableWrite`.